File: backend/services/alpacaClient.js
const axios = require('axios');
require('dotenv').config();

const client = axios.create({
  baseURL: process.env.ALPACA_BASE_URL,
  headers: {
    'APCA-API-KEY-ID': process.env.ALPACA_KEY,
    'APCA-API-SECRET-KEY': process.env.ALPACA_SECRET
  }
});

const getCryptoAssets = async () => {
  const res = await client.get('/v2/assets');
  return res.data.filter(a => a.class === 'crypto' && a.status === 'active');
};

const getBars = async (symbol, timeframe = '1Min', limit = 100) => {
  const res = await client.get(`/v2/stocks/${symbol}/bars`, {
    params: { timeframe, limit }
  });
  return res.data.bars;
};

const placeOrder = async (order) => {
  return client.post('/v2/orders', order);
};

module.exports = { getCryptoAssets, getBars, placeOrder };

File: backend/alpaca_client.py
from alpaca_trade_api.rest import REST
from config import ALPACA_API_KEY, ALPACA_SECRET_KEY, ALPACA_BASE_URL

alpaca = REST(
    key_id=ALPACA_API_KEY,
    secret_key=ALPACA_SECRET_KEY,
    base_url=ALPACA_BASE_URL
)

def get_crypto_assets():
    return [a.symbol for a in alpaca.get_assets(status="active") if a.class_ == "crypto"]

def get_account():
    return alpaca.get_account()

def get_latest_price(symbol):
    barset = alpaca.get_crypto_bars(symbol, timeframe="1Min", limit=1)
    return float(barset[-1].close) if barset else None
